### PR TITLE
Upgrade Rails to 7.1

### PR DESCRIPTION
**Pull request by akhilgkrishnan**
- This PR upgrades the rails version to 7.1

### CODE REVIEW

File: config/initializers/new_framework_defaults_7_1.rb [hunk 1]
```diff
@@ -0,0 +1,224 @@
+# frozen_string_literal: true
```
Comment (akhilgkrishnan) [inline config/initializers/new_framework_defaults_7_1.rb:1]: We can uncomment the changes, that we required.

**Comment by vipulnsward** [comment]
@akhilgkrishnan can you rebase this?

**Comment by akhilgkrishnan** [comment]
> @akhilgkrishnan can you rebase this?

Rebased the PR

**Comment by vipulnsward** [comment]
Specs are failing, @apoorv1316 can you take a look?

File: Gemfile [hunk 2]
```diff
@@ -115,7 +115,7 @@ gem "active_interaction"
 gem "stripe"
 
 # Background job processing adapter
-gem "sidekiq", "<8"
+gem "sidekiq", "~> 7.2"
```
Comment (apoorv1316) [inline Gemfile:118]: Upgraded sidekiq to fix the following error on running sidekiq
`super: no superclass method `broadcast' for ActiveSupport::Logger:Class`

**Comment by apoorv1316** [comment]
Removed search_service and its specs because we are not using ransack for search. It was old code which was not being used.

**Comment by apoorv1316** [comment]
@gowsik-ragunath @prasanthchaduvula @akhilgkrishnan Please review

File: Gemfile.lock [hunk 3]
```diff
@@ -621,11 +655,12 @@ GEM
     websocket-extensions (0.1.5)
     xpath (3.2.0)
       nokogiri (~> 1.8)
-    zeitwerk (2.6.7)
+    zeitwerk (2.6.12)
 
 PLATFORMS
   arm64-darwin-21
   arm64-darwin-22
+  arm64-darwin-23
```
Comment (nisusam) [inline Gemfile.lock:663]: Either get all the relative platforms or remove this

File: db/schema.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema[7.0].define(version: 2024_03_04_143641) do
+ActiveRecord::Schema[7.1].define(version: 2024_03_31_034308) do
```
Comment (nisusam) [inline db/schema.rb:13]: Only single-line change? please check whether the above migrations ran or not.

**Review by nisusam — CHANGES_REQUESTED** [review]

File: Gemfile.lock (hunk 3 above)
Comment (apoorv1316) [reply Gemfile.lock]: done

File: db/schema.rb (hunk 4 above)
Comment (apoorv1316) [reply db/schema.rb:13]: Yes it is up

**Review by nisusam — APPROVED** [review]

**Comment by apoorv1316** [comment]
> @apoorv1316 @nisusam Are we good to merge this PR?

will merge tomorrow

**Pull request merged**